### PR TITLE
Ot208-60

### PR DESCRIPTION
**Pull request by acamus79**
The endpoint of organization/public is modified to obtain a list of Slides of an organization sorted according to the Slide orders field. An important detail is that in the tests it was verified that when uploading an image in S3 of amazon to assign a url to a slide, a copy of the image was generated in the root directory of the project, so the myFile.delete() line is added after uploading the file and before returning the url.

imagen

### CODE REVIEW

**Comment by nestoragredollanten** [comment]
Checked

**Comment by nestoragredollanten** [comment]
Checked

File: src/main/java/com/alkemy/ong/controllers/OrganizationController.java [hunk 1]
```diff
@@ -2,31 +2,42 @@
...
+    private final SlideService slideService;
 
-    @GetMapping("/public")
+/*    @GetMapping("/public")
     public ResponseEntity<OrganizationPublicDto> publicData(@RequestParam String id) {
         OrganizationPublicDto publicDataDTO = organizationService.getPublicOrganizationData(id);
         return ResponseEntity.ok().body(publicDataDTO);
-    }
+    }*/
+@GetMapping("/public")
```
Comment (MartinIriarte89) [inline src/main/java/com/alkemy/ong/controllers/OrganizationController.java:27]: Creo que el endpoint deberia seguir devolviendo los datos de la organizacion, y ademas se deberia agregar el listado de slides a la respuesta.

File: src/main/java/com/alkemy/ong/controllers/OrganizationController.java [hunk 2]
```diff
@@ -2,31 +2,42 @@
...
+public ResponseEntity<List<SlideResponseDto>> publicData(@RequestParam String id) {
+    if (organizationService.existById(id)) {
+        List<SlideResponseDto> publicDataDTO = slideService.getAllByOrganizationId(id);
```
Comment (MartinIriarte89) [inline src/main/java/com/alkemy/ong/controllers/OrganizationController.java:30]: No vi si estaban ordenados por el campo order que pide el ticket cuando se listan, si no esta deberia implementarse.

File: src/main/java/com/alkemy/ong/controllers/OrganizationController.java (hunk 1 above)
Comment (acamus79) [reply src/main/java/com/alkemy/ong/controllers/OrganizationController.java:21]: The interpretation is changed from displaying a list of slides with a common organization to displaying an organization with a list of slides.

File: src/main/java/com/alkemy/ong/controllers/OrganizationController.java (hunk 2 above)
Comment (acamus79) [reply src/main/java/com/alkemy/ong/controllers/OrganizationController.java]: This is solved in the Slides repository by means of Query, SQL takes care of returning it already sorted.

**Review by MartinIriarte89 — APPROVED** [review]
Aprobado

**Pull request merged**